src/Compilers/CSharp/Portable/Symbols/Metadata/PE/NativeIntegerTypeDecoder.cs

```diff
@@ -31,11 +31,24 @@ private static TypeSymbol TransformType(TypeSymbol type, ImmutableArray<bool> tr
                 {
                     return result;
                 }
+                else
+                {
+                    return new UnsupportedMetadataTypeSymbol();
+                }
+            }
+            catch (UnsupportedSignatureContent)
```

Comment:
>UnsupportedSignatureContent [](start = 19, length = 27)

Could we throw `UnsupportedMetadataTypeSymbol` originally to avoid catching and throwing a different exception here?